scripts/build-environment.sh

```diff
@@ -1,10 +1,69 @@
 #!/bin/bash
+usage() {
+    cat << USAGE >&2
+USAGE:
+    $ build-environment.sh [PARAMETERS]
+PARAMETERS:
+    --build-base-bee                    The base bee image will be built from source code
+    --base-bee-commit-hash=string       the source code commit hash of the base bee; Default: HEAD; Dependency: --build-base-bee
+USAGE
+    exit 1
+}
+
+build_bee() {
+    # Clone source code
+    BEE_SOURCE_PATH=$MY_PATH/../bee
+    if [ -d "$BEE_SOURCE_PATH" ] ; then
+        rm -rf "$BEE_SOURCE_PATH"
+    fi
+    mkdir "$BEE_SOURCE_PATH" && cd "$BEE_SOURCE_PATH" || exit 1
+    git init
+    git remote add origin https://github.com/ethersphere/bee.git
+    git fetch origin --depth=1 "$COMMIT_HASH"
+    git reset --hard FETCH_HEAD
```

Comment:
I haven't used this command before, what does it do?

Reply:
You can see an explanation [here](https://stackoverflow.com/a/9237511):

> FETCH_HEAD is a short-lived ref, to keep track of what has just been fetched from the remote repository (...)
FETCH_HEAD points to the tip of this branch (it stores the SHA1 of the commit, just as branches do).